packages/core/src/tools/create-app-request-client.js

```diff
@@ -53,17 +53,14 @@ const createAppRequestClient = (input, options) => {
 
   const httpOriginalAfters = [prepareResponse, logResponse];
 
-  if (app.authentication) {
-    if (
-      app.authentication.type === 'session' ||
-      (app.authentication.type === 'oauth2' &&
-        _.get(app, 'authentication.oauth2Config.autoRefresh'))
-    ) {
-      httpOriginalAfters.push(throwForStaleAuth);
```

Comment:
No more need for this as we now default `afterResponse` to `[throwForStatus]`, which will throw for 401 and on the server-side we check if that should result in a refresh for the app's definition.